src/app/src/pages/Comments.tsx

```diff
@@ -0,0 +1,277 @@
+import { ButtonGroup, ButtonProps, InputLabel, TextField } from "@mui/material";
+import { styled as Styled } from "@mui/material/styles";
+import { useParams } from "react-router-dom";
+
+interface Props {
+  currentUser?: Author;
+}
+
+const EditContainer = styled.div`
+  background-color: white;
+  width: 80%;
+  height: 80%;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  border-radius: 10px;
+`;
+const Block = styled.div`
+  width: 100%;
+  height: 100%;
+  color: black;
+  display: flex;
+  flex-direction: column;
+  overflow-y: scroll;
+  text-align: left;
+  align-items: center;
+`;
+const Header = styled.div`
+  margin-top: 1%;
+  font-size: 200%;
+  text-align: center;
+`;
+
+const Content = styled.div`
+  margin-top: 1%;
+  width: 80%;
+  height: 20%;
+  display: flex;
+  flex-direction: column;
+`;
+
+const ActualContent = styled.div`
+  margin-top: 2%;
+  width: 50%;
+`;
+const WriteOrPreview = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+`;
+
+const CustomButton = Styled(Button)<ButtonProps>(({ theme }) => ({
+  color: theme.palette.getContrastText("#fff"),
+  padding: "10px",
+  backgroundColor: "white",
+  "&:hover": {
+    backgroundColor: "#b5b5b5",
+  },
+}));
+const formStyle = { m: 1, minWidth: 120, width: "40%", mt: 2 };
+
+export default function Comments({ currentUser }: Props): JSX.Element {
+  let { postID, authorID } = useParams();
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [post, setPost] = useState<Post>();
+  const [commentsChanged, setCommentsChanged] = useState(false);
+  const [open, setOpen] = useState(false);
+  const [type, setType] = useState<any>("");
+  const [openWrite, setOpenWrite] = useState(true);
+  const [content, setContent] = useState("");
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleType = (event: SelectChangeEvent) => {
+    setType(event.target.value);
+  };
+  const handleTextChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setContent(event.target.value);
+  };
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        if (authorID && postID) {
+          const commentList = api.authors.withId(authorID).posts.withId(postID).comments.list();
+          const post = api.authors.withId(authorID).posts.withId(postID).get();
+          post.then((data) => setPost(data)).catch((err) => console.log(err));
+          commentList.then((comment) => {
+            setComments(comment);
+            setCommentsChanged(false);
+          });
+        }
+      } catch (err) {
+        console.log(err);
+      }
+    }
+    fetchData();
+  }, [authorID, postID, commentsChanged]);
+
+  const comment = {
+    comment: content,
+    contentType: type,
+  };
+  const createComment = () => {
+    if (authorID && postID) {
+      api.authors
+        .withId(authorID)
```

Comment:
We'd want the currentUser to create the comment. Not the authorID one, that's who'se post we're commenting on.